### ISSUE TITLE
Forcing binding updates

### ISSUE THREAD

**Issue by gen3ric** [SeriousM/WPFLocalizationExtension#91]
I have a unique implementation using a custom JSON provider, where my JSON is downloaded and loaded into a dictionary throughout the application lifecycle. How can I go about letting my `LocBinding`s know that they need to requery?

**Comment by SeriousM** [comment]
You mean you got new localizations and you want to update the bindings?
Am 24.02.2016 22:17 schrieb "Eric Rodewald" <notifications@github.com>:

**Comment by gen3ric** [comment]
Yes, that's correct.

**Comment by SeriousM** [comment]
I guess you implemented ILocalizationProvider, right?
you can rise [ProviderChanged](https://github.com/SeriousM/WPFLocalizationExtension/blob/master/WPFLocalizeExtension/Providers/ILocalizationProvider.cs#L62) to force an update.
Can you verify please?

**Comment by gen3ric** [comment]
@SeriousM Thanks for your response. I realized I was woefully out of date (2.1.2?) because NuGet did not think there was a newer package available. I removed and reinstalled the nupkg and updated my `ILocalizationProvider`. I better understand how it works now, and am correctly using `BLoc` instead of `Loc` in XAML, but it still won't quite sink in for me how I could notify the bindings to update with a new value.

My observation is that `OnProviderChanged(DependencyObject obj)` can be called, but it requires a reference to a specific dependency object. I would not have a reference to that. 

The context of my requirement is when I download a new JSON language file, create the new dictionary and add it to the `JsonLocalizationProvider` that I created. So I would want to force all the bindings  to update _en masse_. 

Am I approaching this entirely wrong?

**Comment by SeriousM** [comment]
@MrCircuit could you help us out?
Am 26.02.2016 03:05 schrieb "Eric Rodewald" <notifications@github.com>:

**Comment by MrCircuit** [comment]
Sorry, but I'm confused. `OnProviderChanged(DependencyObject obj)` is a custom implementation of the ResxLocalizationProviderBase class and is not defined in the interface. Did you use this class as parent for your implementation?

Nevertheless, what's behind the curtain is, that this 'DependencyObject' is later needed for the event arguments of the 'ProviderChanged' event. This in turn will be relayed to all registered 'IDictionaryEventListener' (e.g. the markup extensions) instances in order to let them decide, whether to update or not. Using this information, 'LocExtension' will look up its own visual tree for the given object and update only in case of a hit - **if null was given, it will update anyway**. **The 'BLocExtension' will always update no matter what object was given.**

To sum it up: The 'DependencyObject' you pass there is just used for performance optimization, if you know where in the visual tree the change happened.
